src/login/logind-dbus.c

```diff
@@ -2232,6 +2257,14 @@ static int method_schedule_shutdown(sd_bus_message *message, void *userdata, sd_
         if (r != 0)
                 return r;
 
+        if (!timestamp_is_set(elapse) && m->maintenance_window) {
+                r = calendar_spec_next_usec(m->maintenance_window, now(CLOCK_REALTIME), &elapse);
+                if (r < 0)
+                        return log_error_errno(r, "failed to parse shutdown time: %m");
```

Comment:
this log message is wrong, you already parsed the expression at this point. Instead the calculation of the next elapsing failed.

Please return a recognizable error for the case where the configured calendar expression has no more points in time where it could elapse. i.e. as above define a proper dbus error to catch this.

calendar_spec_next_usec() returns -ENOENT if the expression cannot elapse anymore